kolibri/plugins/coach/assets/src/state/actions/reports.js

```diff
@@ -124,24 +125,69 @@ function _showRecentChannels(store, classId) {
 
 
 function _showRecentReports(store, classId, channelId) {
-  store.dispatch('CORE_SET_PAGE_LOADING', true);
-  store.dispatch('SET_PAGE_NAME', Constants.PageNames.RECENT);
-
   // should be cached if navigated to this point
   const channelPromise = ChannelResource.getModel(channelId).fetch();
+  function __reportPropsGen(report) {
+    function ___descriptor(kind) {
+      switch (kind) {
+        case ContentNodeKinds.AUDIO:
+          return 'Listened';
+        case ContentNodeKinds.DOCUMENT:
+          return 'Opened';
+        case ContentNodeKinds.VIDEO:
+          return 'Watched';
+        case ContentNodeKinds.EXERCISE:
+          return 'Mastered';
+        case ContentNodeKinds.HTML5:
+          return 'Mastered';
+        default:
+          return 'Mastered';
+      }
```

Comment:
Note that all of these values need to be `$tr` wrapped. This will be easier once the logic is moved to a computed prop in a vue.